### PR TITLE
Improve pull request template

### PR DESCRIPTION
**Pull request by johnnyomair**
Extend the template to guide contributors on how to create an easy-to-review pull request.

---

## PR Checklist

-   [x] Link to the respective task if one exists: COM-973

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,14 +1,111 @@
...
+-   An example usage of your change
+
+--->
+
+<!--
+
+Everything below this is intended to help ease reviewing this PR.
+Remove all unrelated sections.
+
+WHEN MERGING THE PR, REMOVE THIS FROM THE COMMIT MESSAGE.
```
Comment (johnnyomair) [inline .github/pull_request_template.md:24]: We could also include everything in the squashed commit message. However, in my experience this section is often incomplete or out of date. What do you think?

**Comment by johnnyomair** [comment]
I'd also like to add a note about unrelated changes and keeping the PRs small. Where should I add it?

**Comment by nsams** [comment]
> I'd also like to add a note about unrelated changes and keeping the PRs small. Where should I add it?

:+1:, maybe add it right to the beginning, as this is probably the most important thing for PRs?

File: .github/pull_request_template.md (hunk 1 above)
Comment (nsams) [reply .github/pull_request_template.md:24]: I think it is perfect as you made it

**Comment by johnnyomair** [comment]
> > I'd also like to add a note about unrelated changes and keeping the PRs small. Where should I add it?
> 
> 👍, maybe add it right to the beginning, as this is probably the most important thing for PRs?

Added it at the beginning, let's hope this is not too late: 1286cc2a91c78d934d9442f304bef8764b62c1a9

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,14 +1,111 @@
...
+Make sure to provide an example of your change if your change includes a new API.
+
+This can be either:
+-   The implementation in Demo
+-   A dev story in Storybook
+-   A unit test
+
+--->
+
+[] I have verified if my change requires an example
```
[the remark this answers is not in the document]
Comment (johnnyomair) [reply .github/pull_request_template.md]: Nice catch, thanks: 8232141cca0d1e499466ee09f6982bb3ab98ffb5

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -1,14 +1,111 @@
...
+-   changing the package's public API (`src/index.ts`)
+-   fixing a bug
+-   making a visual change
+
+Changeset writing guidelines:
+-   Use active voice: "Add new thing" vs. "A new thing is added"
+-   First line should be the title: "Add new alert component"
+-   Provide additional information in the description
+-   Use backticks to highlight code: Add new `Alert` component
+-   Use bold formatting for "headlines": **Example**
```
[the remark this answers is not in the document]
Comment (johnnyomair) [reply .github/pull_request_template.md]: Fixed: 7bf18ba19cd8036e5115f08deda228e50bffa094

**Review by thomasdax98 — APPROVED** [review]
I like it, although it's a lot of text. I think we should test it like this and iterate on it later if it turns out to be too annoying

**Comment by johnnyomair** [comment]
> although it's a lot of text.

I know, but our devs asked for more 😁 IMO this could all be in CONTRIBUTING.md, but apparently nobody reads that...

I used comments so that the text isn't visible in the description.

**Review by jamesricky — APPROVED** [review]

**Pull request merged**